practices/managing-node-projects.md

```diff
@@ -20,78 +20,79 @@
+1. Dependency analysis tools (such as `npm audit` and Dependabot) can spot insecure dependencies anywhere in the dependency tree by analysing the `package-lock.json` file. These tools can force updates of insecure dependencies in the tree, without waiting for third-party package maintainers to release fixes. This wouldn't be possible without a `package-lock.json` and is a huge benefit.
 
 The downside is it requires all developers to use `nvm` and `npm ci` to install dependencies, otherwise there will be constant conflicts in the `package-lock.json` file.
 
 ### Managing changes to the `package-lock.json` file
 
-Firstly, bear in mind machine-generated files should not be hand-edited, and `package-lock.json` is no exception.
+Firstly, bear in mind machine-generated files should not be hand-edited, including `package-lock.json`.
 
 Secondly, if you're seeing unexpected merge conflicts in `package-lock.json` it can be a symptom of someone using `npm install` instead of `npm ci`. Speak to the developer and see if they are having trouble.
 
-When you update your application, we strongly recommend that you keep `package-lock.json` changes as a result of `npm update` or `npm audit fix` in a separate PR to other changes to the application. This helps by making your PRs easier to review.
+When you update your dependencies and need to commit changes to `package-lock.json`, we strongly recommend commiting the minimal amount of changes to keep the application running - i.e. that you are especially careful to practice "[Atomic Commits](https://codesignal.com/learn/courses/git-basics/lessons/commits-in-detail-making-them-atomic-crafting-messages-and-using-amendments)".
 
 ## Specifying versions of dependencies
 
-As a general rule, we tend to follow a conservative approach when specifying versions for the dependencies of our packages. We only allow PATCH version ranges for [run-time dependencies](#run-time-dependencies) and MINOR + PATCH version ranges for [dev dependencies](#development-dependencies). This helps us ensure consistency between different environments, as we can't know for sure in what environment the app is going to be running, and prevent breakage caused by major or minor updates in dependencies.
+As a general rule, we tend to follow a conservative approach when specifying versions for the dependencies of our packages. We recommend PATCH version ranges for [run-time dependencies](#run-time-dependencies) and MINOR + PATCH version ranges for [dev dependencies](#development-dependencies). This helps us ensure consistency between different environments, (as we can't know for sure in what environment the app is going to be running), and helps prevent breakage caused by major or minor updates in dependencies.
```

Comment:
Is this still the case? Considering the actual landscape and the problems we've seen when updating certain npm modules in the past, I'm wondering if we should instead recommend that people pin specific version numbers in order to avoid breakages and improve reproducibility.

Reply:
Right now I think I agree with you. These are the changes I alluded to re. the second half of the document, as our perspective has changed since this doc was written - but I'd like us to discuss internally before suggesting bigger changes in a different PR.

But while I was here, I did want to soften the wording in this section from "only allow" to "recommend"...

Reply:
Excellent!